src/components/UnifedAxis/CombinedYAxis.js

```diff
@@ -0,0 +1,167 @@
+      }
+    }
+  }
+
+  getDomain = seriesArray => {
+    const yDomain = [Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER];
+    seriesArray.forEach(s => {
+      yDomain[0] = Math.min(yDomain[0], s.yDomain[0]);
+      yDomain[1] = Math.max(yDomain[1], s.yDomain[1]);
+    });
+    return yDomain;
+  };
+
+  syncZoomingState = () => {
+    if (this.props.zoomable) {
+      this.selection.call(this.zoom);
+    } else {
+      this.selection.on('.zoom', null);
+    }
+  };
+
+  didZoom = () => {
+    const t = d3.event.transform;
+    this.props.series.forEach(s => {
+      this.props.updateYTransformation(s.id, t, this.props.height);
+    });
+  };
+
+  renderZoomRect() {
+    const { height, width } = this.props;
+    return (
+      <rect
+        width={width}
+        height={height}
+        fill="none"
+        pointerEvents="all"
+        ref={ref => {
+          this.zoomNode = ref;
+        }}
+      />
+    );
+  }
+
+  renderAxis() {
+    const { series, height, color } = this.props;
+    const scale = createYScale(this.getDomain(series), height);
+    const axis = d3.axisRight(scale);
+    const tickFontSize = 14;
+    const strokeWidth = 2;
+    const halfStrokeWidth = strokeWidth / 2;
+    const tickSizeOuter = axis.tickSizeOuter();
+    const tickSizeInner = axis.tickSizeInner();
+    const tickPadding = axis.tickPadding();
+    // same as for xAxis but consider height of the screen ~two times smaller
+    const values = scale.ticks(Math.floor(height / 50) || 1);
+    const k = 1;
+    const tickFormat = scale.tickFormat();
+    const range = scale.range().map(r => r + halfStrokeWidth);
+    const pathString = [
+      // Move to this (x,y); start drawing
+      `M${k * tickSizeOuter},${range[0] - strokeWidth}`,
+      // Draw a horizontal line halfStrokeWidth long
+      `H${halfStrokeWidth}`,
+      // Draw a vertical line from bottom to top
+      `V${range[1]}`,
+      // Finish with another horizontal line
+      `H${k * tickSizeOuter}`,
+    ].join('');
+    return (
+      <g
+        className="axis"
+        fill="none"
+        fontSize={tickFontSize}
+        textAnchor="start"
+        strokeWidth={strokeWidth}
+      >
+        <path stroke={series.color || color} d={pathString} />
+        {values.map(v => {
+          const lineProps = { stroke: series.color || color };
+          lineProps.x2 = k * tickSizeInner;
+          lineProps.y1 = halfStrokeWidth;
+          lineProps.y2 = halfStrokeWidth;
+
+          const textProps = { fill: series.color || color, dy: '0.32em' };
+          textProps.x = k * Math.max(tickSizeInner, 0) + tickPadding;
+          textProps.y = halfStrokeWidth;
+          return (
+            <g key={+v} opacity={1} transform={`translate(0, ${scale(v)})`}>
+              <line {...lineProps} />
+              <text {...textProps}>{tickFormat(v)}</text>
+            </g>
+          );
+        })}
+      </g>
+    );
+  }
+
+  render() {
```

Comment:
It would be super cool if `YAxis` and `UnifiedAxis` common logic will be moved onto their parent component

Reply:
I definitely agree. But .. not yet :-) In fact, the I went ahead and removed most of the common logic (because zooming won't work anyway)

Reply:
Could you make it clear why axis zooming won't work?

Reply:
Sure, I'll add a comment in there explaining it, as best as I understand.